<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Efstathios-Iason Vlachavas">
<meta name="dcterms.date" content="2024-04-21">

<title>ACCC-CRC RNASeq Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="ACCC_CRC_GE_files/libs/clipboard/clipboard.min.js"></script>
<script src="ACCC_CRC_GE_files/libs/quarto-html/quarto.js"></script>
<script src="ACCC_CRC_GE_files/libs/quarto-html/popper.min.js"></script>
<script src="ACCC_CRC_GE_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="ACCC_CRC_GE_files/libs/quarto-html/anchor.min.js"></script>
<link href="ACCC_CRC_GE_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="ACCC_CRC_GE_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="ACCC_CRC_GE_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="ACCC_CRC_GE_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="ACCC_CRC_GE_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

<script src="ACCC_CRC_GE_files/libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="ACCC_CRC_GE_files/libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="ACCC_CRC_GE_files/libs/datatables-binding-0.30/datatables.js"></script>
<script src="ACCC_CRC_GE_files/libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="ACCC_CRC_GE_files/libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="ACCC_CRC_GE_files/libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="ACCC_CRC_GE_files/libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="ACCC_CRC_GE_files/libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet">
<script src="ACCC_CRC_GE_files/libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="ACCC_CRC_GE_files/libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet">
<script src="ACCC_CRC_GE_files/libs/selectize-0.12.0/selectize.min.js"></script>
<link href="ACCC_CRC_GE_files/libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="ACCC_CRC_GE_files/libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<style>
figcaption {
  text-align: center;
}
</style>


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#accc-crc-rnaseq-dataset-analysis-overview" id="toc-accc-crc-rnaseq-dataset-analysis-overview" class="nav-link active" data-scroll-target="#accc-crc-rnaseq-dataset-analysis-overview"><span class="header-section-number">1</span> ACCC-CRC RNASeq dataset analysis overview</a>
  <ul class="collapse">
  <li><a href="#reproduce-the-analysis-of-the-rnaseq-data-from-the-main-accc-crc-dataset-for-the-manuscript." id="toc-reproduce-the-analysis-of-the-rnaseq-data-from-the-main-accc-crc-dataset-for-the-manuscript." class="nav-link" data-scroll-target="#reproduce-the-analysis-of-the-rnaseq-data-from-the-main-accc-crc-dataset-for-the-manuscript."><span class="header-section-number">1.0.1</span> Reproduce the analysis of the rnaseq data from the main ACCC-CRC dataset for the manuscript.</a></li>
  <li><a href="#then-load-preprocess-the-rnaseq-data" id="toc-then-load-preprocess-the-rnaseq-data" class="nav-link" data-scroll-target="#then-load-preprocess-the-rnaseq-data"><span class="header-section-number">1.0.2</span> Then load-preprocess the rnaseq data</a></li>
  <li><a href="#eda-de-analysis" id="toc-eda-de-analysis" class="nav-link" data-scroll-target="#eda-de-analysis"><span class="header-section-number">1.1</span> EDA &amp; DE analysis</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">ACCC-CRC RNASeq Analysis</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Efstathios-Iason Vlachavas </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">2024-04-21</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="accc-crc-rnaseq-dataset-analysis-overview" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> ACCC-CRC RNASeq dataset analysis overview</h1>
<section id="reproduce-the-analysis-of-the-rnaseq-data-from-the-main-accc-crc-dataset-for-the-manuscript." class="level3" data-number="1.0.1">
<h3 data-number="1.0.1" class="anchored" data-anchor-id="reproduce-the-analysis-of-the-rnaseq-data-from-the-main-accc-crc-dataset-for-the-manuscript."><span class="header-section-number">1.0.1</span> Reproduce the analysis of the rnaseq data from the main ACCC-CRC dataset for the manuscript.</h3>
<p>Initially start by loading the necessary R packages, along with importing the clinical information.</p>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># For the appropriate order of jumping into the analysis, please check initially and refer to the Clinical_Process_MSI.R file, to see the respective clinical files and MSI inferred information;</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co"># load the necessary packages and initially read the necessary clinical data:</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EnsDb.Hsapiens.v75)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(biomaRt)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(progeny)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dorothea)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(decoupleR)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(xml2)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(downlit)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(circlize)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hrbrthemes)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="fu">dr_here</span>()</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>merged.final.accc.clin.pheno.dat <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">here</span>(<span class="st">"Data"</span>, <span class="st">"CRC.ACCC.Clinical.Total.Merged.MSI.Mut.Info.RNASeq.tsv"</span>))</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>eda_pheno_accc <span class="ot">&lt;-</span> merged.final.accc.clin.pheno.dat <span class="sc">|&gt;</span></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">select</span>(Sample_Name, Condition, MSI_status, sex, tumor_location)</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a><span class="fu">datatable</span>(eda_pheno_accc, <span class="at">filter =</span> <span class="st">"top"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-4e47950c252e18d02cd0" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-4e47950c252e18d02cd0">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"0\" data-max=\"1\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28"],["CRC_ACCC_01","CRC_ACCC_02","CRC_ACCC_03","CRC_ACCC_04","CRC_ACCC_05","CRC_ACCC_06","CRC_ACCC_07","CRC_ACCC_08","CRC_ACCC_09","CRC_ACCC_11","CRC_ACCC_12","CRC_ACCC_13","CRC_ACCC_14","CRC_ACCC_15","CRC_ACCC_16","CRC_ACCC_17","CRC_ACCC_18","CRC_ACCC_19","CRC_ACCC_20","CRC_ACCC_21","CRC_ACCC_22","CRC_ACCC_23","CRC_ACCC_24","CRC_ACCC_25","CRC_ACCC_26","CRC_ACCC_27","CRC_ACCC_28","CRC_ACCC_29"],["KRAS_mut",null,"RAS_RAF_GNAS_wt","RAS_RAF_GNAS_wt","GNAS_mut","RAS_RAF_GNAS_mut","KRAS_mut","RAS_RAF_GNAS_wt","RAS_RAF_GNAS_mut","BRAF_mut","RAS_RAF_GNAS_wt","KRAS_mut","RAS_RAF_GNAS_wt","RAS_RAF_GNAS_mut","RAS_RAF_GNAS_wt","BRAF_mut","KRAS_mut","NRAS_mut","GNAS_mut","RAS_RAF_GNAS_wt","RAS_RAF_GNAS_wt","RAS_RAF_GNAS_mut","BRAF_mut","RAS_RAF_GNAS_wt","RAS_RAF_GNAS_wt","KRAS_mut","BRAF_mut","RAS_RAF_GNAS_wt"],[0,0,0,0,0,0,0,0,1,0,0,0,0,1,0,1,0,0,0,1,0,0,1,0,0,0,0,0],["F","F","M","F","F","F","M","M","F","M","M","F","M","M","F","M","F","M","F","M","M","M","F","M","F","F","M","F"],["left_sided","left_sided","left_sided","left_sided","right_sided","right_sided","right_sided","right_sided","right_sided","left_sided","left_sided","right_sided","left_sided","right_sided","right_sided","right_sided","left_sided","right_sided","left_sided","left_sided","right_sided","right_sided","right_sided","right_sided","left_sided","left_sided","right_sided","left_sided"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Sample_Name<\/th>\n      <th>Condition<\/th>\n      <th>MSI_status<\/th>\n      <th>sex<\/th>\n      <th>tumor_location<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="then-load-preprocess-the-rnaseq-data" class="level3" data-number="1.0.2">
<h3 data-number="1.0.2" class="anchored" data-anchor-id="then-load-preprocess-the-rnaseq-data"><span class="header-section-number">1.0.2</span> Then load-preprocess the rnaseq data</h3>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># read first the raw counts expression object:</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"Data/ACCC_CRC_RNASeq_Raw_Counts_DT.rda"</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co"># continue with annotation process </span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>edb <span class="ot">&lt;-</span> EnsDb.Hsapiens.v75</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>ensembl.genes <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">rownames</span>(dt.counts))</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>gene.ids <span class="ot">&lt;-</span> ensembldb<span class="sc">::</span><span class="fu">select</span>(edb, <span class="at">keys =</span> ensembl.genes, <span class="at">keytype =</span> <span class="st">"GENEID"</span>,</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"GENENAME"</span>, <span class="st">"GENEBIOTYPE"</span>)) <span class="co"># ensembldb::select; check with updated versions if the bug with mapIDs has been corrected-also regarding the duplicates, by using GENEBIOTYPE instead of TXBIOTYPE there should not be duplicates anymore</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>final.annot <span class="ot">&lt;-</span> gene.ids <span class="sc">|&gt;</span> <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(GENENAME)) <span class="sc">|&gt;</span> </span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">filter</span>(GENEBIOTYPE <span class="sc">==</span> <span class="st">"protein_coding"</span>) <span class="sc">|&gt;</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">distinct</span>(GENENAME,<span class="at">.keep_all=</span><span class="cn">TRUE</span>)</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>sel.ids <span class="ot">&lt;-</span> <span class="fu">as.character</span>(final.annot<span class="sc">$</span>GENEID)</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>dt.counts.sel <span class="ot">&lt;-</span> dt.counts[sel.ids,]</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a><span class="co"># then assign the relative rownames</span></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>merged.final.accc.clin.pheno.dat <span class="ot">&lt;-</span> merged.final.accc.clin.pheno.dat <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">MSI_status_reform=</span><span class="fu">ifelse</span>(MSI_status<span class="sc">==</span><span class="st">"1"</span>,<span class="st">"MSI"</span>,<span class="st">"MSS"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a><span class="fu">column_to_rownames</span>(<span class="st">"Sample_Name"</span>)</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>mat.counts <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(dt.counts.sel)</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(merged.final.accc.clin.pheno.dat), <span class="fu">colnames</span>(mat.counts)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># redundant but can run for sanity check regarding downstream approaches;</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>merged.final.accc.clin.pheno.dat<span class="sc">$</span>MSI_status_reform <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(merged.final.accc.clin.pheno.dat<span class="sc">$</span>MSI_status_reform)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts=</span>mat.counts, <span class="at">samples =</span> merged.final.accc.clin.pheno.dat)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>y<span class="sc">$</span>genes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ENSEMBL=</span><span class="fu">rownames</span>(y), <span class="at">SYMBOL=</span>final.annot<span class="sc">$</span>GENENAME)</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>cond.MSI <span class="ot">&lt;-</span> y<span class="sc">$</span>samples<span class="sc">$</span>MSI_status_reform</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="do">## The function *filterByExpr* keeps genes that have count-per-million (CPM) above k in n samples, where k is determined by min.count and by the sample library sizes and n is determined by the design matrix.</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="do">## n is essentially the smallest group sample size or, more generally, the minimum inverse leverage of any fitted value; In parallel, each retained gene is required to have at least min.total.count reads across all the samples.</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="do">## group argument: vector or factor giving group membership. </span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>keep.exprs <span class="ot">&lt;-</span> <span class="fu">filterByExpr</span>(y, <span class="at">group=</span>cond.MSI)</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>y.filt <span class="ot">&lt;-</span> y[keep.exprs,, keep.lib.sizes<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>final.exp.gene.vector <span class="ot">&lt;-</span> <span class="fu">as.character</span>(y.filt<span class="sc">$</span>genes<span class="sc">$</span>SYMBOL)</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>exp.dt.accc.crc.rnaseq <span class="ot">&lt;-</span> final.exp.gene.vector <span class="sc">|&gt;</span> <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">ExpressedGenes.ACCC.CRC.RNAseq=</span>value)</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a><span class="co"># write_tsv(exp.dt.accc.crc.rnaseq,"Exp.DT.ACCC.CRC.RNAseq.ReClassified.MSI.tsv")</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(y.filt, <span class="at">method =</span> <span class="st">"TMM"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="eda-de-analysis" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="eda-de-analysis"><span class="header-section-number">1.1</span> EDA &amp; DE analysis</h2>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>lcpm <span class="ot">&lt;-</span> <span class="fu">cpm</span>(y, <span class="at">log=</span><span class="cn">TRUE</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(lcpm, <span class="at">las=</span><span class="dv">2</span>, <span class="at">col=</span><span class="st">"red"</span>, <span class="at">main=</span><span class="st">""</span>, <span class="at">names =</span> <span class="fu">rep</span>(<span class="st">""</span>, <span class="fu">ncol</span>(lcpm)))</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="fu">title</span>(<span class="at">main=</span><span class="st">"A: Unnormalised data"</span>,<span class="at">ylab=</span><span class="st">"Log-cpm"</span>) </span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>lcpm <span class="ot">&lt;-</span> <span class="fu">cpm</span>(dge, <span class="at">log=</span><span class="cn">TRUE</span>)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(lcpm, <span class="at">las=</span><span class="dv">2</span>, <span class="at">col=</span><span class="st">"blue"</span>, <span class="at">main=</span><span class="st">""</span>, <span class="at">names =</span> <span class="fu">rep</span>(<span class="st">""</span>, <span class="fu">ncol</span>(lcpm)))</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="fu">title</span>(<span class="at">main=</span><span class="st">"B: Normalised data"</span>,<span class="at">ylab=</span><span class="st">"Log-cpm"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="ACCC_CRC_GE_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid" width="1056"></p>
</div>
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="do">######## PCA plot with multiple labels ###############</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>mat <span class="ot">&lt;-</span> lcpm</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(mat), <span class="at">scale=</span><span class="cn">TRUE</span>) <span class="co"># check about scaling</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>eig <span class="ot">&lt;-</span> (pca<span class="sc">$</span>sdev)<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>variance <span class="ot">&lt;-</span> eig<span class="sc">*</span><span class="dv">100</span><span class="sc">/</span><span class="fu">sum</span>(eig)</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span><span class="fu">as.data.frame</span>(dge<span class="sc">$</span>samples)</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(dat, <span class="fu">aes</span>(pca<span class="sc">$</span>x[,<span class="dv">1</span>], pca<span class="sc">$</span>x[,<span class="dv">2</span>], <span class="at">color=</span><span class="fu">factor</span>(MSI_status), <span class="at">shape=</span>Condition)) <span class="sc">+</span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label=</span>Condition)) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">"PC1: </span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="st">"</span>,<span class="fu">round</span>(variance[<span class="dv">1</span>],<span class="dv">2</span>),<span class="st">"% variance"</span>)) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">"PC2: </span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="st">"</span>,<span class="fu">round</span>(variance[<span class="dv">2</span>],<span class="dv">2</span>),<span class="st">"% variance"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="ACCC_CRC_GE_files/figure-html/unnamed-chunk-3-2.png" class="img-fluid" width="1056"></p>
</div>
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>MSI_status_cond <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(dge<span class="sc">$</span>samples<span class="sc">$</span>MSI_status_reform)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span><span class="dv">0</span> <span class="sc">+</span> MSI_status_cond)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(design) <span class="ot">&lt;-</span> <span class="fu">levels</span>(MSI_status_cond)</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>v1 <span class="ot">&lt;-</span> <span class="fu">voomWithQualityWeights</span>(dge, design, <span class="at">plot =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="ACCC_CRC_GE_files/figure-html/unnamed-chunk-3-3.png" class="img-fluid" width="1056"></p>
</div>
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="do">###############################################################################################</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co"># optional: save here the normalized/transformed counts into a object to be used for downstream visualization purposes</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>norm.voom.mat <span class="ot">&lt;-</span> v1<span class="sc">$</span>E</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="co"># save to use downstream-the same as above when created </span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(norm.voom.mat) <span class="ot">&lt;-</span> <span class="fu">as.character</span>(v1<span class="sc">$</span>genes<span class="sc">$</span>SYMBOL) </span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>accc.crc.norm.dat <span class="ot">&lt;-</span> norm.voom.mat <span class="sc">|&gt;</span> <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span> <span class="fu">rownames_to_column</span>()</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="co"># write_tsv(accc.crc.norm.dat, file="ACCC.CRC.Norm.VoomQualWeights.CPM.REclassified.MSI.tsv")</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="do">##############################################################################################</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>voom.fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(v1, design)</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>contrast.matrix <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="at">comp1=</span> MSI <span class="sc">-</span> MSS,</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">levels=</span>design)   </span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>fit2 <span class="ot">&lt;-</span> <span class="fu">contrasts.fit</span>(voom.fit, contrast.matrix)</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>fit3 <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit2, <span class="at">trend =</span> <span class="cn">TRUE</span>, <span class="at">robust =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSA</span>(fit3, <span class="at">main=</span><span class="st">"Final model: Mean-variance trend"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="ACCC_CRC_GE_files/figure-html/unnamed-chunk-3-4.png" class="img-fluid" width="1056"></p>
</div>
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>comp1_stat <span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">topTable</span>(fit3, <span class="at">coef=</span><span class="dv">1</span>, <span class="at">number=</span><span class="fu">nrow</span>(fit3), <span class="at">adjust.method=</span><span class="st">"fdr"</span>, <span class="at">sort.by=</span><span class="st">"none"</span>) <span class="sc">|&gt;</span> <span class="fu">rownames_to_column</span>() <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">select</span>(SYMBOL,t) <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="st">"SYMBOL"</span>) <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">MSI_vs_MSS_stat=</span>t)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="pathway-activity-analysis-initially-utilize-the-msi-status-information-to-perform-a-similar-analysis-as-in-the-two-public-crc-datasets-part-of-the-output-for-figure-3-creation" class="level4" data-number="1.1.0.1">
<h4 data-number="1.1.0.1" class="anchored" data-anchor-id="pathway-activity-analysis-initially-utilize-the-msi-status-information-to-perform-a-similar-analysis-as-in-the-two-public-crc-datasets-part-of-the-output-for-figure-3-creation"><span class="header-section-number">1.1.0.1</span> Pathway activity analysis-initially utilize the MSI status information, to perform a similar analysis as in the two public CRC datasets {part of the output for Figure 3 creation}</h4>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>total_stat_dat <span class="ot">&lt;-</span> comp1_stat <span class="sc">|&gt;</span> <span class="fu">as.matrix</span>()</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>PathwayActivity_zscore <span class="ot">&lt;-</span> <span class="fu">progeny</span>(total_stat_dat, </span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">scale=</span><span class="cn">TRUE</span>, <span class="at">organism=</span><span class="st">"Human"</span>, <span class="at">top =</span> <span class="dv">100</span>, <span class="at">perm =</span> <span class="dv">10000</span>, <span class="at">z_scores =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">t</span>()</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(PathwayActivity_zscore) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"NES_MSI_vs_MSS"</span>)</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>PathwayActivity_zscore_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(PathwayActivity_zscore) <span class="sc">|&gt;</span> </span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"Pathway"</span>) <span class="sc">|&gt;</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(NES_MSI_vs_MSS) <span class="sc">|&gt;</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Pathway =</span> <span class="fu">factor</span>(Pathway))</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(PathwayActivity_zscore_df,<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Pathway, NES_MSI_vs_MSS), </span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> NES_MSI_vs_MSS)) <span class="sc">+</span> </span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">fill =</span> NES_MSI_vs_MSS), <span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> <span class="st">"darkblue"</span>, <span class="at">high =</span> <span class="st">"indianred"</span>, </span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">mid =</span> <span class="st">"whitesmoke"</span>, <span class="at">midpoint =</span> <span class="dv">0</span>) <span class="sc">+</span> </span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> </span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>            <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size =</span><span class="dv">10</span>, <span class="at">face=</span> <span class="st">"bold"</span>),</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span><span class="dv">10</span>, <span class="at">face=</span> <span class="st">"bold"</span>),</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"Pathways"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="ACCC_CRC_GE_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>Path.act.dat <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(PathwayActivity_zscore)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>Path.act.dat <span class="ot">&lt;-</span> Path.act.dat <span class="sc">|&gt;</span> <span class="fu">rownames_to_column</span>() <span class="sc">|&gt;</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">PROGENY_pathways=</span>rowname)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co"># write_tsv(Path.act.dat, file="PathwayActivity_zscore.ACCC.CRC.ReClassified.MSI.Pathways.tsv")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="also-small-example-to-further-exploit-specific-pathway-responsive-genes-dependencies" class="level4" data-number="1.1.0.2">
<h4 data-number="1.1.0.2" class="anchored" data-anchor-id="also-small-example-to-further-exploit-specific-pathway-responsive-genes-dependencies"><span class="header-section-number">1.1.0.2</span> Also small example to further exploit specific pathway-responsive genes dependencies:</h4>
<p>here we exploit further the TGFb immune pathway</p>
<div class="cell" data-layout-align="center">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>prog_matrix <span class="ot">&lt;-</span> <span class="fu">getModel</span>(<span class="st">"Human"</span>, <span class="at">top=</span><span class="dv">100</span>) <span class="sc">|&gt;</span> </span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">as.data.frame</span>()  <span class="sc">|&gt;</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"GeneID"</span>)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>pd_TGFb <span class="ot">&lt;-</span> prog_matrix <span class="sc">|&gt;</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">abs</span>(TGFb))) <span class="sc">|&gt;</span> <span class="fu">slice_head</span>(<span class="at">n=</span><span class="dv">100</span>)</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>comp_1_df <span class="ot">&lt;-</span> comp1_stat <span class="sc">|&gt;</span> <span class="fu">rownames_to_column</span>()</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>sel.genes.TGFb <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(pd_TGFb, comp_1_df, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"GeneID"</span><span class="ot">=</span><span class="st">"rowname"</span>))</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>sel.genes.TGFb <span class="ot">&lt;-</span> sel.genes.TGFb <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">select</span>(GeneID,TGFb,MSI_vs_MSS_stat)</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>df_TGFb_out <span class="ot">&lt;-</span> sel.genes.TGFb <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">Gene_impact=</span>TGFb<span class="sc">*</span>MSI_vs_MSS_stat) <span class="sc">|&gt;</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">abs</span>(Gene_impact))) <span class="sc">|&gt;</span> <span class="fu">slice_head</span>(<span class="at">n=</span><span class="dv">20</span>)</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>df_TGFb <span class="ot">&lt;-</span> sel.genes.TGFb <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">color =</span> <span class="st">"3"</span>) <span class="sc">|&gt;</span> <span class="fu">mutate</span>(<span class="at">Gene_impact=</span>TGFb<span class="sc">*</span>MSI_vs_MSS_stat) <span class="sc">|&gt;</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">abs</span>(Gene_impact))) <span class="sc">|&gt;</span> <span class="fu">slice_head</span>(<span class="at">n=</span><span class="dv">20</span>) <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">TGFb_weight=</span>TGFb) <span class="sc">|&gt;</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">color =</span> <span class="fu">if_else</span>(TGFb_weight <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> MSI_vs_MSS_stat <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">'1'</span>, color)) <span class="sc">|&gt;</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">color =</span> <span class="fu">if_else</span>(TGFb_weight <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> MSI_vs_MSS_stat <span class="sc">&lt;</span> <span class="dv">0</span>, <span class="st">'2'</span>, color)) <span class="sc">|&gt;</span></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">color =</span> <span class="fu">if_else</span>(TGFb_weight <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> MSI_vs_MSS_stat <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">'2'</span>, color)) <span class="sc">|&gt;</span></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">color =</span> <span class="fu">if_else</span>(TGFb_weight <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> MSI_vs_MSS_stat <span class="sc">&lt;</span> <span class="dv">0</span>, <span class="st">'1'</span>, color))</span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(df_TGFb, <span class="fu">aes</span>(<span class="at">x =</span> TGFb_weight, <span class="at">y =</span> MSI_vs_MSS_stat, <span class="at">color =</span> color)) <span class="sc">+</span> </span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"red"</span>,<span class="st">"royalblue3"</span>,<span class="st">"grey"</span>)) <span class="sc">+</span></span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label_repel</span>(<span class="fu">aes</span>(<span class="at">label =</span> GeneID)) <span class="sc">+</span> </span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">'dotted'</span>) <span class="sc">+</span></span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">'dotted'</span>) <span class="sc">+</span></span>
<span id="cb12-29"><a href="#cb12-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"TGFb"</span>) <span class="sc">+</span></span>
<span id="cb12-30"><a href="#cb12-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_ipsum</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="ACCC_CRC_GE_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption class="figure-caption">The TGFb pathway seems to be de-activated in MSI when compared to MSS, since the majority of top-20 most responsive target genes with positive weights have negative t-values (4th quadrant).</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="similarly-for-the-tf-activity-inference-analysis" class="level4" data-number="1.1.0.3">
<h4 data-number="1.1.0.3" class="anchored" data-anchor-id="similarly-for-the-tf-activity-inference-analysis"><span class="header-section-number">1.1.0.3</span> Similarly for the TF-activity inference analysis</h4>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>plot_top_features <span class="ot">&lt;-</span> <span class="cf">function</span>(data, n_top) {</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(data) <span class="ot">&lt;-</span> <span class="st">"value"</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  arranged <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"id"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(value))</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>  top_up <span class="ot">&lt;-</span> <span class="fu">slice_head</span>(arranged, <span class="at">n =</span> n_top)</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>  top_down <span class="ot">&lt;-</span> <span class="fu">slice_tail</span>(arranged, <span class="at">n =</span> n_top)</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">list</span>(<span class="at">up =</span> top_up, <span class="at">down =</span> top_down), <span class="at">.id =</span> <span class="st">"status"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">id =</span> <span class="fu">fct_inorder</span>(id)) <span class="sc">%&gt;%</span></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> value, <span class="at">y =</span> id, <span class="at">fill =</span> status)) <span class="sc">+</span></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"up"</span> <span class="ot">=</span> <span class="st">"red"</span>, <span class="st">"down"</span> <span class="ot">=</span> <span class="st">"blue"</span>)) <span class="sc">+</span></span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(p)</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>dorothea_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(dorothea_hs[dorothea_hs<span class="sc">$</span>confidence <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"A"</span>,<span class="st">"B"</span>,<span class="st">"C"</span>),<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">4</span>)])</span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>dorothea_df<span class="sc">$</span>likelihood <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>dorothea_df <span class="ot">&lt;-</span> <span class="fu">intersect_regulons</span>(<span class="fu">as.matrix</span>(comp1_stat), dorothea_df, <span class="at">.source =</span> tf, <span class="at">.target =</span> target, <span class="at">minsize=</span><span class="dv">20</span>)</span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>TF_activities_MSI_vs_MSS <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">run_wmean</span>(<span class="at">mat =</span> <span class="fu">as.matrix</span>(comp1_stat), <span class="at">network =</span> dorothea_df, <span class="at">.source =</span> <span class="st">"tf"</span>, <span class="at">.target =</span> <span class="st">"target"</span>, <span class="at">times =</span> <span class="dv">1000</span>))</span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a><span class="co"># for extra save with all information:</span></span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>dat.TF_activities_MSI_vs_MSS <span class="ot">&lt;-</span> TF_activities_MSI_vs_MSS <span class="sc">|&gt;</span> </span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">filter</span>(statistic<span class="sc">==</span><span class="st">"norm_wmean"</span>) <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">select</span>(statistic,source,score,p_value)</span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a><span class="co"># write_tsv(dat.TF_activities_MSI_vs_MSS, file="ACCC.CRC.TFact.MSI_vs_MSS.RES.TotalTable.ReClassified.tsv")</span></span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a>TF_activities_MSI_vs_MSS <span class="ot">&lt;-</span> TF_activities_MSI_vs_MSS[TF_activities_MSI_vs_MSS<span class="sc">$</span>statistic <span class="sc">==</span> <span class="st">"norm_wmean"</span>,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">4</span>)] <span class="sc">|&gt;</span> <span class="fu">as_tibble</span>() <span class="sc">|&gt;</span> <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">"source"</span>)</span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_top_features</span>(TF_activities_MSI_vs_MSS, <span class="at">n_top =</span> <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">'TF top10 activities MSI_vs_MSS'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="ACCC_CRC_GE_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="also-continue-with-firstly-creating-the-figure-4.a-part-with-pathwaytf-activities-per-patient" class="level4" data-number="1.1.0.4">
<h4 data-number="1.1.0.4" class="anchored" data-anchor-id="also-continue-with-firstly-creating-the-figure-4.a-part-with-pathwaytf-activities-per-patient"><span class="header-section-number">1.1.0.4</span> Also continue with firstly creating the <strong>Figure 4.A</strong> part with pathway/TF activities per patient</h4>
<div class="cell">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(circlize)</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(InteractiveComplexHeatmap)</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>out.CMS.ACCC <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">here</span>(<span class="st">"Data"</span>,<span class="st">"Out_RES_DT_CMScaller_ACCC_28samples.tsv"</span>))</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>out.CMS.ACCC.sel <span class="ot">&lt;-</span> out.CMS.ACCC <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(ACCC_Sample_ID, prediction)</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"C:/Users/e376s/Desktop/OnkoKB.Match.Scored.ACCC/ACCC_CRC_Patient_Level/Clustering/FINAL"</span>)</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>cluster_info <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">here</span>(<span class="st">"Data"</span>,<span class="st">"HCPC_FactoMineR_RES_ACCC_CRC.tsv"</span>))</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>cluster_info_merge <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(cluster_info, out.CMS.ACCC.sel, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"Sample_ID"</span><span class="ot">=</span><span class="st">"ACCC_Sample_ID"</span>))</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>cluster_info_ord <span class="ot">&lt;-</span> cluster_info_merge[<span class="fu">order</span>(cluster_info_merge<span class="sc">$</span>clust, cluster_info_merge<span class="sc">$</span>MSI_status, cluster_info_merge<span class="sc">$</span>Condition),] <span class="co"># updated grouping based also on subsequently the MSI status + mutational groups</span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>pdat_ord <span class="ot">&lt;-</span> cluster_info_ord <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="st">"Sample_ID"</span>,<span class="st">"MSI_status"</span>,<span class="st">"Condition"</span>,<span class="st">"sex"</span>, <span class="st">"histological_subtype"</span>,<span class="st">"tumor_location"</span>,<span class="st">"clust"</span>,<span class="st">"prediction"</span>)) <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="st">"Sample_ID"</span>)</span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a><span class="co"># load the patient-inferred pathway/TF activities:</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>Path_act <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">here</span>(<span class="st">"Data"</span>,</span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a><span class="st">"PathwayActivity_zscore_ACCC_CRC_Scaled_PatientLevel_PROGENY.tsv"</span>))</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>Path_act_2 <span class="ot">&lt;-</span> Path_act <span class="sc">|&gt;</span> <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">"PROGENY_pathways"</span>)</span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>TF_act <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">here</span>(<span class="st">"Data"</span>,<span class="st">"TFactivity_zscore_ACCC_CRC_Scaled_PatientLevel_DOROTHEA.tsv"</span>))</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>TF_act_2 <span class="ot">&lt;-</span> TF_act <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">select</span>(TFs_Dorothea,condition,score) <span class="sc">|&gt;</span> <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"condition"</span>,<span class="at">values_from=</span><span class="st">"score"</span>) <span class="sc">|&gt;</span> <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">"TFs_Dorothea"</span>)</span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a><span class="co"># reorder columns for each numeric dataset</span></span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>pathway_scaled_order <span class="ot">=</span> Path_act_2[, <span class="fu">rownames</span>(pdat_ord)]</span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>TF_scaled_order<span class="ot">=</span> TF_act_2[,<span class="fu">rownames</span>(pdat_ord)]</span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>col_fun_1 <span class="ot">=</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">0</span>, <span class="dv">5</span>), <span class="fu">c</span>(<span class="st">"#377EB8"</span>, <span class="st">"white"</span>, <span class="st">"#E41A1C"</span>))</span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>ha_add <span class="ot">=</span> <span class="fu">HeatmapAnnotation</span>(</span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">MSI_status =</span> <span class="fu">ifelse</span>(pdat_ord[[<span class="dv">1</span>]] <span class="sc">==</span> <span class="dv">1</span>, <span class="st">"MSI"</span>,<span class="st">"MSS"</span>),</span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">Mutation_group =</span> pdat_ord[[<span class="dv">2</span>]],</span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sex =</span> <span class="fu">ifelse</span>(pdat_ord[[<span class="dv">3</span>]] <span class="sc">==</span> <span class="st">"F"</span>, <span class="st">"Female"</span>,<span class="st">"Male"</span>),</span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">Histo_Subtype =</span> pdat_ord[[<span class="dv">4</span>]],</span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">Tumor_Location =</span> pdat_ord[[<span class="dv">5</span>]],</span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">Cluster_ID =</span> pdat_ord[[<span class="dv">6</span>]],</span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">CMS_subtype =</span> pdat_ord[[<span class="dv">7</span>]],</span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">border =</span> <span class="cn">TRUE</span>,</span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">list</span>(<span class="at">Mutation_group =</span> <span class="fu">structure</span>(<span class="at">names =</span> <span class="fu">c</span>(<span class="st">"BRAF_mut"</span>, <span class="st">"KRAS_mut"</span>, </span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a>    <span class="st">"GNAS_mut"</span>, <span class="st">"NRAS_mut"</span>, <span class="st">"RAS_RAF_GNAS_wt"</span>, <span class="st">"RAS_RAF_GNAS_mut"</span>), </span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"yellow"</span>, <span class="st">"red"</span>, <span class="st">"black"</span>, <span class="st">"green"</span>, <span class="st">"orange"</span>, <span class="st">"cyan"</span>)),</span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a>    <span class="at">MSI_status =</span> <span class="fu">structure</span>(<span class="at">names =</span> <span class="fu">c</span>(<span class="st">"MSI"</span>,<span class="st">"MSS"</span>), <span class="fu">c</span>(<span class="st">"red"</span>,<span class="st">"black"</span>)),</span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sex =</span> <span class="fu">structure</span>(<span class="at">names =</span> <span class="fu">c</span>(<span class="st">"Female"</span>,<span class="st">"Male"</span>), <span class="fu">c</span>(<span class="st">"red"</span>,<span class="st">"black"</span>)),</span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a>    <span class="at">Histo_Subtype =</span> <span class="fu">structure</span>(<span class="at">names =</span> <span class="fu">c</span>(<span class="st">"adenoCa_int_diff"</span>,<span class="st">"adenoCa_low_diff"</span>,<span class="st">"adenoCa_NOS"</span>,</span>
<span id="cb14-49"><a href="#cb14-49" aria-hidden="true" tabindex="-1"></a>    <span class="st">"desmin_adenoCa"</span>,<span class="st">"LVI_adenoCa_int diff"</span>,<span class="st">"src_mucinous_adenoCa"</span>), </span>
<span id="cb14-50"><a href="#cb14-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"yellow"</span>, <span class="st">"red"</span>, <span class="st">"blue"</span>, <span class="st">"black"</span>, <span class="st">"green"</span>, <span class="st">"orange"</span>)),</span>
<span id="cb14-51"><a href="#cb14-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">Tumor_Location =</span> <span class="fu">structure</span>(<span class="at">names =</span> <span class="fu">c</span>(<span class="st">"left_sided"</span>,<span class="st">"right_sided"</span>), <span class="fu">c</span>(<span class="st">"red"</span>,<span class="st">"black"</span>)),</span>
<span id="cb14-52"><a href="#cb14-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">Cluster_ID =</span> <span class="fu">structure</span>(<span class="at">names =</span> <span class="fu">c</span>(<span class="st">"1"</span>,<span class="st">"2"</span>,<span class="st">"3"</span>),<span class="fu">c</span>(<span class="st">"black"</span>,<span class="st">"red"</span>,<span class="st">"blue"</span>)),</span>
<span id="cb14-53"><a href="#cb14-53" aria-hidden="true" tabindex="-1"></a>    <span class="at">CMS_subtype =</span> <span class="fu">structure</span>(<span class="at">names =</span> <span class="fu">c</span>(<span class="st">"CMS1"</span>,<span class="st">"CMS2"</span>,<span class="st">"CMS3"</span>,<span class="st">"CMS4"</span>),            <span class="fu">c</span>(<span class="st">"orange"</span>,<span class="st">"blue"</span>,<span class="st">"purple"</span>,<span class="st">"darkgreen"</span>)))) </span>
<span id="cb14-54"><a href="#cb14-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-55"><a href="#cb14-55" aria-hidden="true" tabindex="-1"></a>ht1_final_2 <span class="ot">=</span> <span class="fu">Heatmap</span>(pathway_scaled_order, <span class="at">name=</span><span class="st">"Pathway_activity"</span>,<span class="at">cluster_columns =</span> <span class="cn">FALSE</span>, <span class="at">top_annotation =</span> ha_add,<span class="at">show_column_names =</span> <span class="cn">TRUE</span>,<span class="at">col=</span>col_fun_1,<span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">9</span>),<span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>, <span class="at">fontface =</span> <span class="st">"bold"</span>), <span class="at">row_title =</span> <span class="st">"Pathways"</span>,</span>
<span id="cb14-56"><a href="#cb14-56" aria-hidden="true" tabindex="-1"></a><span class="at">show_row_dend =</span> <span class="cn">FALSE</span>,</span>
<span id="cb14-57"><a href="#cb14-57" aria-hidden="true" tabindex="-1"></a><span class="at">cell_fun =</span> <span class="cf">function</span>(j, i, x, y, w, h, fill) {</span>
<span id="cb14-58"><a href="#cb14-58" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(pathway_scaled_order[i, j] <span class="sc">&gt;=</span> <span class="dv">2</span>) {</span>
<span id="cb14-59"><a href="#cb14-59" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grid.text</span>(<span class="st">"*"</span>, x, y)</span>
<span id="cb14-60"><a href="#cb14-60" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span>(pathway_scaled_order[i, j] <span class="sc">&lt;=</span> <span class="sc">-</span><span class="dv">2</span>) {</span>
<span id="cb14-61"><a href="#cb14-61" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grid.text</span>(<span class="st">"*"</span>, x, y)</span>
<span id="cb14-62"><a href="#cb14-62" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb14-63"><a href="#cb14-63" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb14-64"><a href="#cb14-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-65"><a href="#cb14-65" aria-hidden="true" tabindex="-1"></a>ht2_final_2 <span class="ot">=</span> <span class="fu">Heatmap</span>(TF_scaled_order, <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>, <span class="at">name=</span> <span class="st">"TF_activity"</span>,</span>
<span id="cb14-66"><a href="#cb14-66" aria-hidden="true" tabindex="-1"></a><span class="at">show_column_names =</span> <span class="cn">TRUE</span>, <span class="at">col=</span>col_fun_1,<span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">9</span>),<span class="at">row_title=</span><span class="st">"TFs"</span>, <span class="at">show_row_dend =</span> <span class="cn">FALSE</span>, <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">8</span>, <span class="at">fontface =</span> <span class="st">"bold"</span>),</span>
<span id="cb14-67"><a href="#cb14-67" aria-hidden="true" tabindex="-1"></a><span class="at">cell_fun =</span> <span class="cf">function</span>(j, i, x, y, w, h, fill) {</span>
<span id="cb14-68"><a href="#cb14-68" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(TF_scaled_order[i, j] <span class="sc">&gt;=</span> <span class="dv">2</span>) {</span>
<span id="cb14-69"><a href="#cb14-69" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grid.text</span>(<span class="st">"*"</span>, x, y)</span>
<span id="cb14-70"><a href="#cb14-70" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> <span class="cf">if</span>(TF_scaled_order[i, j] <span class="sc">&lt;=</span> <span class="sc">-</span><span class="dv">2</span>) {</span>
<span id="cb14-71"><a href="#cb14-71" aria-hidden="true" tabindex="-1"></a>        <span class="fu">grid.text</span>(<span class="st">"*"</span>, x, y)</span>
<span id="cb14-72"><a href="#cb14-72" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb14-73"><a href="#cb14-73" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb14-74"><a href="#cb14-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-75"><a href="#cb14-75" aria-hidden="true" tabindex="-1"></a>ht_list <span class="ot">=</span> ht1_final_2 <span class="sc">%v%</span> ht2_final_2</span>
<span id="cb14-76"><a href="#cb14-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-77"><a href="#cb14-77" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(ht_list)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="ACCC_CRC_GE_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" width="300"></p>
</div>
</div>
<button type="button" class="btn btn-primary" data-bs-toggle="modal" data-bs-target="#exampleModal">
<p>See in Full Screen</p>
</button>
<div id="exampleModal" class="modal fade" tabindex="-1">
<div class="modal-dialog modal-dialog-centered p-8" style="max-width: 90%">
<div class="modal-content">
<div class="cell">
<div class="cell-output-display">
<p><img src="ACCC_CRC_GE_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" style="width:100.0%"></p>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="then-conceptualize-the-figure-4.b-part-for-the-merged-visualization-with-the-ssgseaimmune-deconvolution-estimates" class="level4" data-number="1.1.0.5">
<h4 data-number="1.1.0.5" class="anchored" data-anchor-id="then-conceptualize-the-figure-4.b-part-for-the-merged-visualization-with-the-ssgseaimmune-deconvolution-estimates"><span class="header-section-number">1.1.0.5</span> Then conceptualize the <strong>Figure 4.B</strong> part for the merged visualization with the ssGSEA/immune deconvolution estimates</h4>
<div class="cell" data-layout-align="center">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(circlize)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>ssGSEA.scores <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">here</span>(<span class="st">"Data"</span>,<span class="st">"ACCC_CRC_ssGSEA_NES_scores.txt"</span>))</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>ssgsea.names <span class="ot">&lt;-</span> <span class="fu">as.character</span>(ssGSEA.scores<span class="sc">$</span>id)</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>ssgsea.names.filt <span class="ot">&lt;-</span> <span class="fu">str_remove</span>(ssgsea.names, <span class="at">pattern =</span> <span class="st">"HALLMARK_"</span>)</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>scored.dat <span class="ot">&lt;-</span> ssGSEA.scores <span class="sc">|&gt;</span> <span class="fu">column_to_rownames</span>(<span class="st">"id"</span>)</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(scored.dat) <span class="ot">&lt;-</span> ssgsea.names.filt</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a><span class="co"># use the same phenotypic information as from above code chunk</span></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a><span class="fu">exists</span>(<span class="st">"pdat_ord"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># continue with the visualization heatmap</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>cluster.ord <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">rownames</span>(pdat_ord))</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>sel.hallmark.terms <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ANGIOGENESIS"</span>,<span class="st">"TNFA_SIGNALING_VIA_NFKB"</span>,</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>               <span class="st">"NOTCH_SIGNALING"</span>,<span class="st">"EPITHELIAL_MESENCHYMAL_TRANSITION"</span>,<span class="st">"GLYCOLYSIS"</span>,</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>               <span class="st">"DNA_REPAIR"</span>,<span class="st">"G2M_CHECKPOINT"</span>,</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>               <span class="st">"PI3K_AKT_MTOR_SIGNALING"</span>,<span class="st">"INFLAMMATORY_RESPONSE"</span>,</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>               <span class="st">"FATTY_ACID_METABOLISM"</span>,<span class="st">"MYC_TARGETS_V1"</span>,<span class="st">"MYC_TARGETS_V2"</span>,</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>               <span class="st">"E2F_TARGETS"</span>,<span class="st">"WNT_BETA_CATENIN_SIGNALING"</span>,</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>               <span class="st">"INTERFERON_GAMMA_RESPONSE"</span>,<span class="st">"UNFOLDED_PROTEIN_RESPONSE"</span>,</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>               <span class="st">"CHOLESTEROL_HOMEOSTASIS"</span>) <span class="co"># selected representative hallmarks</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>gsea.score.sel <span class="ot">&lt;-</span> scored.dat[sel.hallmark.terms,cluster.ord]</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>NES_ssGEA <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(gsea.score.sel))))</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(NES_ssGEA)[<span class="dv">4</span>] <span class="ot">&lt;-</span> <span class="st">"EMT"</span> <span class="co"># FOR "EPITHELIAL_MESENCHYMAL_TRANSITION"</span></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a><span class="co"># create an additional heatmap with the ESTIMATE scores</span></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>ESTIMATE.dat <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">here</span>(<span class="st">"Data"</span>,</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a><span class="st">"Immunedeconv.RES.ESTIMATE.ACCC.CRC.MSI.REClassified.UnLogTPM.tsv"</span>))</span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>ESTIMATE.dat.filt <span class="ot">&lt;-</span> ESTIMATE.dat <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(ESTIMATE_Scores<span class="sc">%in%</span><span class="fu">c</span>(<span class="st">"StromalScore"</span>,<span class="st">"ImmuneScore"</span>)) <span class="sc">|&gt;</span> <span class="fu">column_to_rownames</span>(<span class="st">"ESTIMATE_Scores"</span>) <span class="sc">|&gt;</span> <span class="fu">as.matrix</span>()</span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>ESTIMATE.mat <span class="ot">&lt;-</span> ESTIMATE.dat.filt[,<span class="fu">rownames</span>(pdat_ord)]</span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a>ESTIMATE_scaled <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(ESTIMATE.mat))))</span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a><span class="co"># also include CIBERSORTx scaled cell fractions</span></span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a>cbx.dat <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="fu">here</span>(<span class="st">"Data"</span>,<span class="st">"CIBERSORTx_Job6_Results_ABSMode.txt"</span>))</span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a><span class="co"># only to fetch the absolute score</span></span>
<span id="cb17-35"><a href="#cb17-35" aria-hidden="true" tabindex="-1"></a>cbx.mat2 <span class="ot">&lt;-</span> cbx.dat <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Mixture,<span class="st">`</span><span class="at">Absolute score (sig.score)</span><span class="st">`</span>) <span class="sc">|&gt;</span> <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="sc">-</span>Mixture) <span class="sc">|&gt;</span></span>
<span id="cb17-36"><a href="#cb17-36" aria-hidden="true" tabindex="-1"></a><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Mixture, <span class="at">values_from =</span> value) <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="st">"name"</span>) <span class="sc">|&gt;</span> <span class="fu">as.matrix</span>()</span>
<span id="cb17-37"><a href="#cb17-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-38"><a href="#cb17-38" aria-hidden="true" tabindex="-1"></a>CIBERSORT_ord <span class="ot">&lt;-</span> cbx.mat2[,cluster.ord]</span>
<span id="cb17-39"><a href="#cb17-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-40"><a href="#cb17-40" aria-hidden="true" tabindex="-1"></a>CIBERSORT_ord_mat <span class="ot">&lt;-</span> CIBERSORT_ord <span class="sc">|&gt;</span> <span class="fu">enframe</span>() <span class="sc">|&gt;</span> <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> name, <span class="at">values_from =</span> value) <span class="sc">|&gt;</span> <span class="fu">as.matrix</span>()</span>
<span id="cb17-41"><a href="#cb17-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-42"><a href="#cb17-42" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(CIBERSORT_ord_mat) <span class="ot">&lt;-</span> <span class="st">"CIBERSORTx_ABS_score"</span></span>
<span id="cb17-43"><a href="#cb17-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-44"><a href="#cb17-44" aria-hidden="true" tabindex="-1"></a>CIBERSORT_ord_mat_scaled <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(CIBERSORT_ord_mat)))) <span class="co"># final numeric input</span></span>
<span id="cb17-45"><a href="#cb17-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-46"><a href="#cb17-46" aria-hidden="true" tabindex="-1"></a><span class="co"># define the color palettes for the heatmaps</span></span>
<span id="cb17-47"><a href="#cb17-47" aria-hidden="true" tabindex="-1"></a>col_fun_1 <span class="ot">=</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">1</span>), <span class="fu">c</span>(<span class="st">"#377EB8"</span>, <span class="st">"white"</span>, <span class="st">"#E41A1C"</span>))</span>
<span id="cb17-48"><a href="#cb17-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-49"><a href="#cb17-49" aria-hidden="true" tabindex="-1"></a>ha_add <span class="ot">=</span> <span class="fu">HeatmapAnnotation</span>(</span>
<span id="cb17-50"><a href="#cb17-50" aria-hidden="true" tabindex="-1"></a>    <span class="at">MSI_status =</span> <span class="fu">ifelse</span>(pdat_ord[[<span class="dv">1</span>]] <span class="sc">==</span> <span class="dv">1</span>, <span class="st">"MSI"</span>,<span class="st">"MSS"</span>),</span>
<span id="cb17-51"><a href="#cb17-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">Mutation_group =</span> pdat_ord[[<span class="dv">2</span>]],</span>
<span id="cb17-52"><a href="#cb17-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sex =</span> <span class="fu">ifelse</span>(pdat_ord[[<span class="dv">3</span>]] <span class="sc">==</span> <span class="st">"F"</span>, <span class="st">"Female"</span>,<span class="st">"Male"</span>),</span>
<span id="cb17-53"><a href="#cb17-53" aria-hidden="true" tabindex="-1"></a>    <span class="at">Histo_Subtype =</span> pdat_ord[[<span class="dv">4</span>]],</span>
<span id="cb17-54"><a href="#cb17-54" aria-hidden="true" tabindex="-1"></a>    <span class="at">Tumor_Location =</span> pdat_ord[[<span class="dv">5</span>]],</span>
<span id="cb17-55"><a href="#cb17-55" aria-hidden="true" tabindex="-1"></a>    <span class="at">Cluster_ID =</span> pdat_ord[[<span class="dv">6</span>]],</span>
<span id="cb17-56"><a href="#cb17-56" aria-hidden="true" tabindex="-1"></a>    <span class="at">CMS_subtype =</span> pdat_ord[[<span class="dv">7</span>]],</span>
<span id="cb17-57"><a href="#cb17-57" aria-hidden="true" tabindex="-1"></a>    <span class="at">border =</span> <span class="cn">TRUE</span>,</span>
<span id="cb17-58"><a href="#cb17-58" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">list</span>(<span class="at">Mutation_group =</span> <span class="fu">structure</span>(<span class="at">names =</span> <span class="fu">c</span>(<span class="st">"BRAF_mut"</span>, <span class="st">"KRAS_mut"</span>, </span>
<span id="cb17-59"><a href="#cb17-59" aria-hidden="true" tabindex="-1"></a>    <span class="st">"GNAS_mut"</span>, <span class="st">"NRAS_mut"</span>, <span class="st">"RAS_RAF_GNAS_wt"</span>, <span class="st">"RAS_RAF_GNAS_mut"</span>), </span>
<span id="cb17-60"><a href="#cb17-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"yellow"</span>, <span class="st">"red"</span>, <span class="st">"black"</span>, <span class="st">"green"</span>, <span class="st">"orange"</span>, <span class="st">"cyan"</span>)),</span>
<span id="cb17-61"><a href="#cb17-61" aria-hidden="true" tabindex="-1"></a>    <span class="at">MSI_status =</span> <span class="fu">structure</span>(<span class="at">names =</span> <span class="fu">c</span>(<span class="st">"MSI"</span>,<span class="st">"MSS"</span>), <span class="fu">c</span>(<span class="st">"red"</span>,<span class="st">"black"</span>)),</span>
<span id="cb17-62"><a href="#cb17-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">Sex =</span> <span class="fu">structure</span>(<span class="at">names =</span> <span class="fu">c</span>(<span class="st">"Female"</span>,<span class="st">"Male"</span>), <span class="fu">c</span>(<span class="st">"red"</span>,<span class="st">"black"</span>)),</span>
<span id="cb17-63"><a href="#cb17-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">Histo_Subtype =</span> <span class="fu">structure</span>(<span class="at">names =</span> <span class="fu">c</span>(<span class="st">"adenoCa_int_diff"</span>,<span class="st">"adenoCa_low_diff"</span>,<span class="st">"adenoCa_NOS"</span>,</span>
<span id="cb17-64"><a href="#cb17-64" aria-hidden="true" tabindex="-1"></a>    <span class="st">"desmin_adenoCa"</span>,<span class="st">"LVI_adenoCa_int diff"</span>,<span class="st">"src_mucinous_adenoCa"</span>), </span>
<span id="cb17-65"><a href="#cb17-65" aria-hidden="true" tabindex="-1"></a>    <span class="fu">c</span>(<span class="st">"yellow"</span>, <span class="st">"red"</span>, <span class="st">"blue"</span>, <span class="st">"black"</span>, <span class="st">"green"</span>, <span class="st">"orange"</span>)),</span>
<span id="cb17-66"><a href="#cb17-66" aria-hidden="true" tabindex="-1"></a>    <span class="at">Tumor_Location =</span> <span class="fu">structure</span>(<span class="at">names =</span> <span class="fu">c</span>(<span class="st">"left_sided"</span>,<span class="st">"right_sided"</span>), <span class="fu">c</span>(<span class="st">"red"</span>,<span class="st">"black"</span>)),</span>
<span id="cb17-67"><a href="#cb17-67" aria-hidden="true" tabindex="-1"></a>    <span class="at">Cluster_ID =</span> <span class="fu">structure</span>(<span class="at">names =</span> <span class="fu">c</span>(<span class="st">"1"</span>,<span class="st">"2"</span>,<span class="st">"3"</span>),<span class="fu">c</span>(<span class="st">"black"</span>,<span class="st">"red"</span>,<span class="st">"blue"</span>)),</span>
<span id="cb17-68"><a href="#cb17-68" aria-hidden="true" tabindex="-1"></a>    <span class="at">CMS_subtype =</span> <span class="fu">structure</span>(<span class="at">names =</span> <span class="fu">c</span>(<span class="st">"CMS1"</span>,<span class="st">"CMS2"</span>,<span class="st">"CMS3"</span>,<span class="st">"CMS4"</span>),            <span class="fu">c</span>(<span class="st">"orange"</span>,<span class="st">"blue"</span>,<span class="st">"purple"</span>,<span class="st">"darkgreen"</span>)))) </span>
<span id="cb17-69"><a href="#cb17-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-70"><a href="#cb17-70" aria-hidden="true" tabindex="-1"></a>ht1 <span class="ot">=</span> <span class="fu">Heatmap</span>(ESTIMATE_scaled, <span class="at">name=</span><span class="st">"Immune_Deconv_ESTIMATE_Zscore"</span>,<span class="at">top_annotation =</span> ha_add,</span>
<span id="cb17-71"><a href="#cb17-71" aria-hidden="true" tabindex="-1"></a>              <span class="at">row_title =</span> <span class="st">"EST"</span>, <span class="at">column_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">7</span>),</span>
<span id="cb17-72"><a href="#cb17-72" aria-hidden="true" tabindex="-1"></a>              <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>, <span class="at">show_column_names =</span> <span class="cn">TRUE</span>, <span class="at">col=</span>col_fun_1, </span>
<span id="cb17-73"><a href="#cb17-73" aria-hidden="true" tabindex="-1"></a>              <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">9</span>),<span class="at">show_row_dend =</span> <span class="cn">FALSE</span>)</span>
<span id="cb17-74"><a href="#cb17-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-75"><a href="#cb17-75" aria-hidden="true" tabindex="-1"></a><span class="co"># this for the absolute ABS score estimate from CIBERSORTX !!</span></span>
<span id="cb17-76"><a href="#cb17-76" aria-hidden="true" tabindex="-1"></a>ht2 <span class="ot">=</span> <span class="fu">Heatmap</span>(CIBERSORT_ord_mat_scaled, <span class="at">name=</span><span class="st">"Immune_Deconv_CIBERSORTx_ABS_Zscore"</span>,</span>
<span id="cb17-77"><a href="#cb17-77" aria-hidden="true" tabindex="-1"></a>              <span class="at">row_title =</span> <span class="st">"CBRx"</span>,<span class="at">cluster_columns =</span> <span class="cn">FALSE</span>, <span class="at">show_column_names =</span> <span class="cn">TRUE</span>,                         <span class="at">col=</span>col_fun_1, <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">9</span>),<span class="at">show_row_dend =</span> <span class="cn">FALSE</span>)</span>
<span id="cb17-78"><a href="#cb17-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-79"><a href="#cb17-79" aria-hidden="true" tabindex="-1"></a>ht3 <span class="ot">=</span> <span class="fu">Heatmap</span>(NES_ssGEA, <span class="at">name=</span><span class="st">"NES_ssGSEA_Zscore"</span>,</span>
<span id="cb17-80"><a href="#cb17-80" aria-hidden="true" tabindex="-1"></a>              <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>, <span class="at">show_column_names =</span> <span class="cn">TRUE</span>, </span>
<span id="cb17-81"><a href="#cb17-81" aria-hidden="true" tabindex="-1"></a>              <span class="at">col=</span>col_fun_1, <span class="at">row_names_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">9</span>),</span>
<span id="cb17-82"><a href="#cb17-82" aria-hidden="true" tabindex="-1"></a><span class="at">row_title =</span> <span class="st">"MsigDb_Sel_Hallmarks"</span>,<span class="at">show_row_dend =</span> <span class="cn">FALSE</span>)</span>
<span id="cb17-83"><a href="#cb17-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-84"><a href="#cb17-84" aria-hidden="true" tabindex="-1"></a>ht_list <span class="ot">=</span> ht1 <span class="sc">%v%</span> ht2 <span class="sc">%v%</span> ht3</span>
<span id="cb17-85"><a href="#cb17-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-86"><a href="#cb17-86" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(ht_list)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="ACCC_CRC_GE_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="1152"></p>
<figcaption class="figure-caption">Heatmap visualization of the normalized enrichment scores of selected hallmarks from MsigDb per patient using ssGSEA.</figcaption>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>

</main>
<!-- /main column -->
&nbsp;
<hr>
<p style="text-align: center;">A work by <a href="https://github.com/Jasonmbg/">Efstathios-Iason Vlachavas</a></p>
<p style="text-align: center;"><span style="color: #808080;"><em>svlachavas@eie.gr</em></span></p>

<!-- Add icon library -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">

<!-- Add font awesome icons -->
<p style="text-align: center;">
    <a href="https://orcid.org/0000-0002-3432-9053" class="fa fa-orcid"></a>
    <a href="https://www.linkedin.com/in/efstathios-iason-vlachavas-164b6085/" class="fa fa-linkedin"></a>
    <a href="https://github.com/Jasonmbg/" class="fa fa-github"></a>
</p>

&nbsp;
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>